core/src/executor/evaluate/expr.rs

```diff
@@ -49,9 +49,19 @@ pub fn binary_op<'a>(
         BinaryOperator::Eq => cmp!(l.evaluate_eq(&r)),
         BinaryOperator::NotEq => cmp!(!l.evaluate_eq(&r)),
         BinaryOperator::Lt => cmp!(l.evaluate_cmp(&r) == Some(Ordering::Less)),
-        BinaryOperator::LtEq => cmp!(l.evaluate_cmp(&r) != Some(Ordering::Greater)),
+        BinaryOperator::LtEq => {
+            if l.evaluate_cmp(&r).is_none() {
+                return Ok(Evaluated::from(Value::Bool(false)));
+            }
+            cmp!(l.evaluate_cmp(&r) != Some(Ordering::Greater))
+        }
```

Comment:
`matches!` would be useful. same approach can be applied for `BinaryOperator::GtEq`
```rust
BinaryOperator::LtEq => cmp!(matches!(
    l.evaluate_cmp(&r),
    Some(Ordering::Less) | Some(Ordering::Equal)
)),
```

Reply:
Wow! matches! macro thank you~